### PR TITLE
Improve error reporting in pq package

### PR DESCRIPTION
**Pull request by urso**
- introduce Error type that is compatible to txerr
- improve support for checking and recovering from root cause by
  introducing error kinds and re-export error kinds
- add more context like queue ID (based on pointer address) and page id
  to errors
- create trace of errors annotated with operations that failed

### CODE REVIEW

File: pq/pq.go [hunk 1]
```diff
@@ -71,53 +82,47 @@ func MakeRoot() [SzRoot]byte {
...
+
 	if delegate == nil {
-		return nil, errNODelegate
+		return nil, errOp(op).of(InvalidParam).report("delegate must not be nil")
 	}
 
-	accessor, err := makeAccess(delegate)
-	if err != nil {
-		return nil, err
+	accessor, k := makeAccess(delegate)
```
Comment (kvch) [inline pq/pq.go]: I would write `errKind` instead of `k`. It's not obvious what's the second variable `makeAccess` return.

File: pq/ack.go [hunk 2]
```diff
@@ -250,10 +248,12 @@ func (a *acker) collectFreePages(c *txCursor, endID uint64) ([]txfile.PageID, bo
...
 // findNewStartPositions skips acked events, so to find the new head and read pointers to be set
 // in the updated queue header.
-func (a *acker) findNewStartPositions(c *txCursor, id uint64) (head, read position, err error) {
+func (a *acker) findNewStartPositions(c *txCursor, id uint64) (head, read position, err reason) {
+	const op = "pq/ack-compute-new-start"
+
 	var hdr *eventPage
 
-	hdr, err = c.PageHeader()
+	hdr, err = c.PageHeader(op)
```
Comment (ph) [inline pq/ack.go]: Why cannot simply `errWrap()` instead of passing the current op?

Comment (ph) [reply pq/ack.go]: I am not sure I understand the need to pass around the current operation to other method? I did a quick look at the other methods receiving the operation and I think we could still wrap in the caller instead?

Comment (urso) [reply pq/ack.go]: Ah, will change it to use errWrap. Code passing `op` is somewhat older, before I introduced errWrap and context merging.

**Pull request merged**